### PR TITLE
feat: manual sealing ✨

### PR DESCRIPTION
**Pull request by EvolveArt**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Build-related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: N/A

# What is the new behavior?

- Implement the `sc-consensus-manual-sealing` pallet.
- Add a `--sealing` option to the `cli`
- Add a new `DevChainSpec` chain spec type

# Does this introduce a breaking change?

- [x] Yes
- [ ] No

# Other information

Highly inspired by the work done by [frontier](https://github.com/paritytech/frontier/blob/master/template/node/src/chain_spec.rs#L27)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -53,12 +53,15 @@ sp-version = { git = "https://github.com/paritytech/substrate", branch = "polkad
...
 sp-storage = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.40", default-features = false }
+sp-state-machine = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.40", default-features = false }
+sp-trie = { version = "7.0.0", git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
 
 # Substrate client dependencies
 sc-client-db = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.40" }
 sc-network = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.40" }
 sc-network-common = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.40" }
 sc-consensus = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.40" }
+sc-consensus-manual-seal = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.40", default-features = false }
```
Comment (abdelhamidbakhta) [inline Cargo.toml:64]: please add a comment saying this is for testing purpose for RPC endpoints and why we need this

File: crates/node/src/chain_spec.rs [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
+#[derive(Serialize, Deserialize)]
+pub struct DevGenesisExt {
+    /// Genesis config.
+    genesis_config: GenesisConfig,
+    /// The flag that if enable manual-seal mode.
+    enable_manual_seal: Option<bool>,
+}
+
+impl sp_runtime::BuildStorage for DevGenesisExt {
+    fn assimilate_storage(&self, storage: &mut Storage) -> Result<(), String> {
```
Comment (abdelhamidbakhta) [inline crates/node/src/chain_spec.rs:36]: add rust doc

Comment (abdelhamidbakhta) [reply crates/node/src/chain_spec.rs:36]: also if you can ELI5 why it's needed it would be great

File: crates/node/src/rpc/mod.rs [hunk 3]
```diff
@@ -27,6 +29,8 @@ pub struct FullDeps<C, P> {
     pub pool: Arc<P>,
     /// Whether to deny unsafe calls
     pub deny_unsafe: DenyUnsafe,
+    /// Manual seal command sink
```
Comment (abdelhamidbakhta) [inline crates/node/src/rpc/mod.rs:32]: what does command sink mean ?

File: crates/runtime/src/config.rs [hunk 4]
```diff
@@ -80,3 +80,7 @@ parameter_types! {
     pub const SS58Prefix: u8 = 42;
     pub FeeMultiplier: Multiplier = Multiplier::one();
 }
+
+parameter_types! {
```
Comment (abdelhamidbakhta) [inline crates/runtime/src/config.rs:84]: explain what is this config used for

File: crates/runtime/src/pallets.rs [hunk 5]
```diff
@@ -150,3 +153,13 @@ impl pallet_utility::Config for Runtime {
     type WeightInfo = ();
     type PalletsOrigin = OriginCaller;
 }
+
+pub struct ConsensusOnTimestampSet<T>(PhantomData<T>);
+impl<T: pallet_aura::Config> OnTimestampSet<T::Moment> for ConsensusOnTimestampSet<T> {
+    fn on_timestamp_set(moment: T::Moment) {
```
Comment (abdelhamidbakhta) [inline crates/runtime/src/pallets.rs:159]: rust doc

**Review by abdelhamidbakhta — CHANGES_REQUESTED** [review]
left some comments

File: crates/node/src/rpc/mod.rs (hunk 3 above)
Comment (EvolveArt) [reply crates/node/src/rpc/mod.rs:32]: un robinet de commandes a.k.a a stream

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm
